src/Microsoft.Graph/Generated/callrecords/requests/SessionRequest.cs

```diff
@@ -154,6 +217,56 @@ public async System.Threading.Tasks.Task<Session> UpdateAsync(Session sessionToU
             return updatedEntity;
         }
 
+        /// <summary>
+        /// Updates the specified Session using PATCH and returns a <see cref="GraphResponse{Session}"/> object.
+        /// </summary>
+        /// <param name="sessionToUpdate">The Session to update.</param>
+        /// <returns>The <see cref="GraphResponse{Session}"/> object of the request.</returns>
+        public System.Threading.Tasks.Task<GraphResponse<Session>> UpdateResponseAsync(Session sessionToUpdate)
+        {
+            return this.UpdateResponseAsync(sessionToUpdate, CancellationToken.None);
+        }
+
+        /// <summary>
+        /// Updates the specified Session using PATCH and returns a <see cref="GraphResponse{Session}"/> object.
+        /// </summary>
+        /// <param name="sessionToUpdate">The Session to update.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> for the request.</param>
+        /// <exception cref="Microsoft.Graph.ClientException">Thrown when an object returned in a response is used for updating an object in Microsoft Graph.</exception>
+        /// <returns>The <see cref="GraphResponse{Session}"/> object of the request.</returns>
+        public async System.Threading.Tasks.Task<GraphResponse<Session>> UpdateResponseAsync(Session sessionToUpdate, CancellationToken cancellationToken)
+        {
+			if (sessionToUpdate.AdditionalData != null)
+			{
+				if (sessionToUpdate.AdditionalData.ContainsKey(Microsoft.Graph.Constants.HttpPropertyNames.ResponseHeaders) ||
```

Comment:
could we imagine moving that control logic into the core SDK instead to avoid duplicating it?

Reply:
That makes sense. We could probably use reflection since the Core library doesn't really know what the object type really is.

Reply:
The optimization could also be applied to the `UpdateAsync` methods that appear just above these calls.

Reply:
I java I have an "IJsonBackedObject" interface that materializes the additional data manager and a few other things. Don't you have the equivalent in the dotnet SDK? in which case type inference would do the trick as far as I can tell from the code